### PR TITLE
My Site Dashboard - Phase 1 - Fix QSP for the Site Picker and Login Epilogue Screens

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #15130 

This PR fixes QSP for the `Site Picker` and `Login Epilogue` screens, in terms of:
- `Site Picker` screen:
  - Making sure that the QSP* is shown at the end of the flow.
  - Making sure that the new QSP* actions, like `Show me around` and `No thanks` are connected to the `Improved My Site` screen. At this point a toast is shown to verify that connection.
- `Login Epilogue` screen:
  - Making sure that when selecting a site, the new `QSP*` actions, like `Show me around` and `No thanks` are connected to the `Improved My Site` screen. At this point a toast is shown to verify that connection.
  - Making sure that when clicking the `Create New Site` button and going through the flow, the new `QSP*` actions, like `Show me around` and `No thanks` are connected to the `Improved My Site` screen. At this point a toast is shown to verify that connection.

PS: In a subsequent PR, the toast functionality will be replaced with the actual implementation, that is the showing or hiding of the `QS*` within the `Improved My Site` screen.

`QS* -> Quick Start`
`QSP* -> Quick Start Prompt`

-----

To test:

Scenario: `Site Picker - Create New Site`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `MySiteImprovementsFeatureConfig` feature flag.
3. Go to the `Site Picker` screen and add a new `wp.com` site.
4. Go through the flow and at the end of it make sure the new `QSP*` is shown.
5. Click the `Show me around` button.
6. Make sure the `Improved My Site` screen is shown and that the `QS - Positive Clicked` toast is shown.
7. Try the same flow, but this time click the `No thanks` button, then make sure the `QS - Negative Clicked` toast is shown.

Scenario: `Login Epilogue - Select Site`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `MySiteImprovementsFeatureConfig` feature flag.
3. Log-out and log-in to the app again.
4. When on the `Login Epilogue` screen, select a site from the list of sites.
5. Make sure the new `QSP*` is shown.
6. Click the `Show me around` button.
7. Make sure the `Improved My Site` screen is shown and that the `QS - Positive Clicked` toast is shown.
8. Try the same flow, but this time click the `No thanks` button, then make sure the `QS - Negative Clicked` toast is shown.

Scenario: `Login Epilogue - Create New Site`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `MySiteImprovementsFeatureConfig` feature flag.
3. Log-out and log-in to the app again.
4. When on the `Login Epilogue` screen, click on the `Create New Site` button.
5. Go through the flow and at the end of it make sure the new `QSP*` is shown.
6. Click the `Show me around` button.
7. Make sure the `Improved My Site` screen is shown and that the `QS - Positive Clicked` toast is shown.
8. Try the same flow, but this time click the `No thanks` button, then make sure the `QS - Negative Clicked` toast is shown.

-----

## Regression Notes
1. Potential unintended areas of impact

The existing `Quick Start` logic (old or/and new) might be somehow affected by all the changes in this PR since there has been some updates to all that.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt [hunk 1]
```diff
@@ -1019,6 +1024,74 @@ class MySiteViewModelTest : BaseUnitTest() {
...
+        whenever(quickStartDynamicCardsFeatureConfig.isEnabled()).thenReturn(false)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(null)
+
+        viewModel.startQuickStart(siteId)
+
+        assertThat(navigationActions).isEmpty()
+    }
+
+    @Test
+    fun `given QS is not available for the site, when quick start start is triggered, then QSP is not shown`() {
```
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: Minor: Typo duplicate `start` here and in the following test case can be removed.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt [hunk 2]
```diff
@@ -281,6 +289,19 @@ class ImprovedMySiteFragment : Fragment(R.layout.new_my_site_fragment),
...
+                R.string.quick_start_dialog_need_help_button_positive,
+                R.string.quick_start_dialog_need_help_manage_site_button_negative,
+                R.string.quick_start_dialog_need_help_button_neutral
+        )
+        is ShowQuickStartDialogNew -> showQuickStartDialog(
+                R.string.quick_start_dialog_need_help_manage_site_title,
+                R.string.quick_start_dialog_need_help_manage_site_message,
+                R.string.quick_start_dialog_need_help_manage_site_button_positive,
+                R.string.quick_start_dialog_need_help_button_negative
+        )
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt]: Question ❓:  How about these strings are set on the navigation action in the VM itself? Then write a test case that the appropriate dialog is shown with the right set of strings?

**Review by ashiagr — APPROVED** [review]
Thank you for the detailed instructions Petros, it was very easy to review the PR 🙌 . 

Approving the PR as everything worked as described, but not merged it so that you can respond to a question and a minor typo comment that I left.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: Thanks for noticing that @ashiagr ! 🙏 

Actually, this was done on purpose, not a typo, although it seems to... 😄 We call the feature `Quick Start` and then we call the function that triggers that `start` + `feature`, so in plain English it becomes `quick start start`. Do you still think that removing the `start` is better?

PS: Same for the below.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt (hunk 2 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt]: Such a nice suggestion @ashiagr , on it!

I should start moving everything to view model, including strings, I sometimes miss that and thanks goodness you are here to help me. 🌟  🙏

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: Ohhh 😯! I didn't realize it 🤦‍♀️. Thanks for clarifying 👍. 

However, I see that you used two different when clauses:

1. `when quick start start is triggered`

2. `when start quick start is triggered`

I like 2. as it doesn't look like a typo 😄, and also makes test statements consistent. Wdyt? Or is that also on purpose 🤔.

Comment (ParaskP7) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: 😄 Oh, I didn't realise I used two different when clauses, thanks for noticing that! 🥇 

Yes, I am going to pick `2` as well and update these two tests, thanks so much @ashiagr ! 🙏

Comment (ParaskP7) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: This is now done here: 6b4f8ec1c64bf52b733425db3b2789db7c699400

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt (hunk 2 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt]: This is now done here: 632894a8dd809f590c97d815799f3a78ea669dd7

**Comment by ashiagr** [comment]
I've reviewed the changes, all looks great! Feel free to merge when Buildkite is done.
